src/datafusion/scanner.rs

```diff
@@ -185,7 +185,10 @@ mod tests {
         let schema = first_batch.schema();
 
         let names = schema.fields().iter().map(|f| f.name()).collect::<Vec<_>>();
-        assert_eq!(names, vec!["lon", "lat"]);
+        let target_names = ["lon", "lat", "float_data"];
```

Comment:
So, in the case or zarr stores, there is no clear order for the "columns", unlike e.g. a parquet file. So while this test as is might have run fine when you tried it, when I did it was picking up "lat" and "float_data", I think it might actually be non-deterministic? Do you think that this could potentially be a problem? When multiple record batches are combined, we could maybe run into issues? If that's the case I could enforce that columns are read in alphabetical order, that would fix it.

Reply:
Sorry about that, but yeah, would be good to give it some order or way to make it deterministic between schema inference and scanning.

Reply:
no worries, we found an issue with a test, that's part of the process! I'll add a fix to this PR, probably tomorrow, and I'll update some tests to explicitly check for column order in the output.